h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -73,6 +73,13 @@ public boolean hasNAs(){
     return false;
   }
 
+  public boolean hasInfs(){
+    // return if frame contains positive infinity
+    for(Vec v:_vecs)
+      if(v.pinfs() > 0) return true;
```

Comment:
There is also ninfs() for negative infinity, shouldn't you check for those as well?
Also as a precaution, you should start rollups on the whole frame before accessing them. You want to avoid rollups to be computed sequentially. You want to do for(Vec v:bulkRollups()) on line 78 instead.